BDArmory/Control/MissileFire.cs

```diff
@@ -6641,7 +6653,7 @@ public bool HasWeaponsAndAmmo(List<WeaponClasses> weaponClasses = null)
                 else { hasWeaponsAndAmmo = true; break; } // Other weapon types don't have ammo, or use electric charge, which could recharge.
             }
             outOfAmmo = !hasWeaponsAndAmmo; // Set outOfAmmo if we don't have any guns with compatible ammo.
-            if (BDArmorySettings.DEBUG_WEAPONS) Debug.Log($"[BDArmory.MissileFire]: {vessel.vesselName} has run out of ammo!");
+            if (BDArmorySettings.DEBUG_WEAPONS && !BDArmorySettings.INFINITE_AMMO) Debug.Log($"[BDArmory.MissileFire]: {vessel.vesselName} has run out of ammo!"); 
```

Comment:
I was getting the "[vessename] has run out of ammo" debug logspam with Infinite Ammo on (test case was a test craft with a gun and infinite ammo, but no ammo boxes), hence the addition.
Re-write looks good.
Infinite Electricity for lasers and railguns and such is already coverd by Infinite Ammo, but yeah, we could add a separate Inf EC if you want infinite lasers, but still have to worry about ammo for railguns/other ECPerShot weapons

Reply:
One reason I was thinking that the infinite EC would be useful is because I was debugging #485 earlier and the plane I was using didn't generate enough EC with its Titan Turboprop engine to keep the radar powered. :-)

Reply:
Added an Infinite EC toggle option

Reply:
Cool. I've updated `HasWeaponsAndAmmo` with what I wrote above locally, btw.